internal/ir/xds.go

```diff
@@ -638,6 +639,13 @@ type CustomResponse struct {
 
 	// StatusCode will be used for the response's status code.
 	StatusCode *uint32 `json:"statusCode,omitempty"`
+
+	// ResponseHeaderModifier defines headers to add, set or remove from the response.
+	// This allows the response policy to append, add or override headers
+	// of the original response before it is sent to a downstream client.
+	//
+	// +optional
+	ResponseHeaderModifier *gwapiv1.HTTPHeaderFilter `json:"responseHeaderModifier,omitempty"`
```

Comment:
can we keep the logic and data types consistent b/w gateway, ir and xds with how they are on Route
https://github.com/envoyproxy/gateway/blob/55bd448cba1be22d8fcaecc3aba40b729c93e50d/internal/ir/xds.go#L775